launcher/models/data_writer.go

```diff
@@ -0,0 +1,46 @@
+// Package models contains models needed in client and server
+package models
+
+import (
+	"fmt"
+	"os"
+	"path"
+)
+
+const (
+	tokenFileTmp = ".token.tmp"
+)
+
+// DataWriter is an interface for writing opaque data to some destination.
+type DataWriter interface {
+	Write(token []byte) error
+}
+
+// FileWriter is a tokenWriter that writes the token to a file.
+type FileWriter struct {
+	directory string
+	filename  string
+}
+
+// Write writes the data to a tmp file before copying it over to the desired location.
+func (t *FileWriter) Write(token []byte) error {
```

Comment:
This write function is specific to write a token file, but the file name (data_writer) is very generic, might want to change the name